apps/nextjs-app/src/features/app/blocks/view/grid/GridViewBase.tsx

```diff
@@ -635,10 +637,11 @@ export const GridViewBase: React.FC<IGridViewProps> = (props: IGridViewProps) =>
 
   useEffect(() => {
     if (!inPrefilling) return;
+    setPrefillingFieldValueMap(extractDefaultFieldsFromFilters(view?.filter));
```

Comment:
I think it might not be the best approach to handle this here.
 You should call `extractDefaultFieldsFromFilters` in `onRowAppend` to get filter-related values, you could also merge them with group-related values, then pass to `generateRecord` for record creation.